### PR TITLE
circulation: allows multiple reminders and overdue fees

### PR DESCRIPTION
**Pull request by zannkukai**
* Updates the `CirculationPolicy` resource JSON schema to allow multiple
  reminders and incremental overdue fees.
* Updates data & fixtures.
* Updates unit test.
* Adapts/improves related code.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: data/circulation_policies.json [hunk 1]
```diff
@@ -9,9 +9,21 @@
...
     "allow_requests": true,
-    "number_of_days_before_due_date": 5,
-    "number_of_days_after_due_date": 5,
-    "reminder_fee_amount": 2.0,
+    "reminders": [
+      {
+        "type": "due_soon",
+        "days_delay": 5,
+        "communication_channel": "email",
+        "template": "email/due_soon"
```
Comment (lauren-d) [inline data/circulation_policies.json:17]: maybe email could be remove and use only `due_soon`, `overdue` or other template name.
it's possible to define template path depending of communication_language

File: rero_ils/modules/notifications/dispatcher.py [hunk 2]
```diff
@@ -80,12 +82,11 @@ def send_mail(data):
...
-        notification_type = data.get('notification_type')
         language = patron['patron']['communication_language']
-        template = 'email/{type}/{lang}.txt'.format(
-            type=notification_type,
-            lang=language
-        )
+        notification_type = data.get('notification_type')
+        loan = Loan.get_record_by_pid(data['loan']['pid'])
+        tpl_path = get_template_to_use(loan, notification_type).rstrip('/')
+        template = f'{tpl_path}/{language}.txt'
```
Comment (lauren-d) [inline rero_ils/modules/notifications/dispatcher.py:89]: see first comment for template path.

cosmetic: .format() ?

File: data/circulation_policies.json (hunk 1 above)
Comment (BadrAly) [reply data/circulation_policies.json:17]: `+ 1` by language by library is possible later

File: rero_ils/modules/circ_policies/api.py [hunk 3]
```diff
@@ -85,124 +89,160 @@ def extended_validation(self, **kwargs):
...
-                        settings__patron_type__pid=patron_type_pid
-                    ),
-                    Q(
-                        'match',
-                        settings__item_type__pid=item_type_pid
-                    )
-                ]
-            )
-        ).source('pid').scan()
+    def get_circ_policy_by_LPI(cls, org_pid, lib_pid, ptty_pid, itty_pid):
```
[the remark this answers is not in the document]
Comment (zannkukai) [reply rero_ils/modules/circ_policies/api.py:161]: it was the name before my changes (just change some code indentation)

**Review by BadrAly — APPROVED** [review]

File: data/circulation_policies.json (hunk 1 above)
Comment (zannkukai) [reply data/circulation_policies.json:17]: for language, this is not the library language but the patron "communication_language" that is used.
So for this point, it isn't the CIPO to choose which template_language_file to use. 
So the actual solution seems woking fine : in `template` we specified the directory where to find the correct `template_language_file.txt`

**Pull request merged**